### PR TITLE
[Notifications Refresh] Replace tabs with dropdown in notifications screen

### PR DESCRIPTION
**Pull request by salimbraksa**
Fixes https://github.com/Automattic/wordpress-mobile/issues/9

| Light | Dark |
| ----- | ----- |
| CleanShot 2024-04-24 at 10 15 48@2x |  CleanShot 2024-04-24 at 10 16 02@2x |

## Test Instructions

1. Run the app and login.
2. Navigate to the Notifications screen
3. Click on the title in the navigation bar
4. It should display a dropdown menu
5. Click on the menu options
6. The dropdown should be dismissed, and then switch to the filter category you clicked, the title should be updated.

## Regression Notes
1. Potential unintended areas of impact
Make sure the correct filter is applied when selecting dropdown options.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual testing.

3. What automated tests I added (or what prevented me from doing so)
None.

## PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding unit tests for my changes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

## Testing checklist:
- [x] WordPress.com sites and self-hosted Jetpack sites.
- [x] Portrait and landscape orientations.
- [x] Light and dark modes.
- [x] Fonts: Larger, smaller and bold text.
- [x] High contrast.
- [x] VoiceOver.
- [x] Languages with large words or with letters/accents not frequently used in English.
- [x] Right-to-left languages. (Even if translation isn’t complete, formatting should still respect the right-to-left layout)
- [x] iPhone and iPad. 
- [x] Multi-tasking: Split view and Slide over. (iPad)

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">This PR is still a Draft: some checks will be skipped.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Review by justtwago — APPROVED** [review]
Looks amazing! It's a pure pleasure to use this and the Notifications tab looks really fresh in its final appearance! 👏 

_I also updated the GH issue to include the update about removing icons from the menu items._

File: WordPress/Classes/ViewRelated/Notifications/Controllers/NotificationsViewController/NotificationsViewController+Strings.swift [hunk 1]
```diff
@@ -19,6 +19,11 @@ extension NotificationsViewController {
                 value: "Navigation Bar Menu Button",
                 comment: "Accessibility label for the navigation bar menu button"
             )
+            static let allNotificationsTitle = NSLocalizedString(
+                "notificationsViewController.navigationBar.filters.allNotifications",
```
Comment (alpavanoglu) [inline WordPress/Classes/ViewRelated/Notifications/Controllers/NotificationsViewController/NotificationsViewController+Strings.swift]: Does it make sense to use `notificationsViewController` as key here? Wouldn't we be better of naming this without technical details, something like `notifications.header.filters.all.notifications`?

Comment (salimbraksa) [reply WordPress/Classes/ViewRelated/Notifications/Controllers/NotificationsViewController/NotificationsViewController+Strings.swift]: Resolved in 96ddcb78d17a9f1a4f03cf2cc2d112da46d2a722. I didn't replace "navigationBar" with "headers" though, as I think "navigationBar" is not too much technical.

**Review by alpavanoglu — APPROVED** [review]

**Pull request merged**